code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -49,15 +76,70 @@
 /obj/item/reagent_containers/food/drinks/cans/afterattack(obj/target, mob/user, proximity)
 	if(!proximity)
 		return
-	if(istype(target, /obj/structure/reagent_dispensers) && (canopened == 0))
+	if(istype(target, /obj/structure/reagent_dispensers) && !canopened)
 		to_chat(user, "<span class='notice'>You need to open the drink!</span>")
 		return
-	else if(target.is_open_container() && (canopened == 0))
+	else if(target.is_open_container() && !canopened)
 		to_chat(user, "<span class='notice'>You need to open the drink!</span>")
 		return
 	else
 		return ..(target, user, proximity)
 
+/obj/item/reagent_containers/food/drinks/cans/after_throw(mob/user)
+	if(shaken < 5)
+		shaken++
+	else
+		bursting(user, TRUE)
+
+/obj/item/reagent_containers/food/drinks/cans/proc/fizzy_open(mob/user, burstopen = FALSE)
+	playsound(loc, 'sound/effects/canopenfizz.ogg', rand(10, 50), 1)
+	canopened = TRUE
+	flags |= OPENCONTAINER
+
+	if(!burstopen)
+		to_chat(user, "<span class='notice'>You open the drink with an audible pop!</span>")
+	else
+		visible_message("<span class='warning'>[src] bursts open!</span>")
+
+	if(shaken < 5)
+		visible_message("<span class='warning'>[src] fizzes violently!</span>")
+	else
+		visible_message("<span class='boldwarning'>[src] erupts into foam!</span>")
+		if(reagents.total_volume)
+			var/datum/effect_system/foam_spread/sodafizz = new
+			sodafizz.set_up(1, get_turf(src), reagents)
+			sodafizz.start()
+
+	for(var/mob/living/carbon/C in range(1, get_turf(src)))
+		to_chat(C, "<span class='warning'>You are splattered with [name]!</span>")
+		reagents.reaction(C, REAGENT_TOUCH)
+		C.wetlevel = max(C.wetlevel + 1, shaken)
+
+	reagents.remove_any((shaken / 5) * reagents.total_volume)
+
+/obj/item/reagent_containers/food/drinks/cans/proc/bursting(mob/user)
```

Comment:
Best to prevent naming procs and variables the same. `bursting` as a proc also reads as if calling this would always burst the can. Which is not the case here